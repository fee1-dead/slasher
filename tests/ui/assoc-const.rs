#![deny(redetect::unused)]

// assoc const usage is also used

trait MyTrait {
    const A: u32;
}

impl MyTrait for u32 {
    const A: u32 = 32;
}
impl MyTrait for u8 {
    //~^ ERROR: implementation of MyTrait for u8 is unused
    const A: u32 = 8;
}

fn main() {
    println!("{}", u32::A);
}
