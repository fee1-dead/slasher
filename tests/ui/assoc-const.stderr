error: implementation of MyTrait for u8 is unused
  --> tests/ui/assoc-const.rs:12:1
   |
12 | impl MyTrait for u8 {
   | ^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/assoc-const.rs:1:9
   |
 1 | #![deny(redetect::unused)]
   |         ^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

